interface/main/calendar/add_edit_event.php

```diff
@@ -977,18 +977,25 @@ function setEventDate($start_date, $recurrence)
     }
 
  // If we have a patient ID, get the name and phone numbers to display.
-    if ($patientid) {
-        $prow = sqlQuery("SELECT lname, fname, phone_home, phone_biz, DOB " .
-         "FROM patient_data WHERE pid = ?", array($patientid));
-        $patientname = $prow['lname'] . ", " . $prow['fname'];
-        if ($prow['phone_home']) {
-            $patienttitle['phone_home'] = xl("Home Phone").": " . $prow['phone_home'];
-        }
+ //dh-3-23-2018 added patient_type to SQL and set the $default_cat_id to the one in patient_data table
+ //need error checking added for case where patient_type doesnt exist in patient_data, or cat doesnt match
+ if ($patientid) {
+    $prow = sqlQuery("SELECT lname, fname, phone_home, phone_biz, DOB, patient_type " .
+     "FROM patient_data WHERE pid = ?", array($patientid));
+    $patientname = $prow['lname'] . ", " . $prow['fname'];
+    //dh adding this section
+    $arow = sqlQuery("Select pc_cat_id from openemr_postcalendar_categories where " .
+    "pc_constant_id = ?", array($prow['patient_type']));
+    $default_cat_id = $arow['patient_type'];
+    //dh
+    if ($prow['phone_home']) {
+        $patienttitle['phone_home'] = xl("Home Phone").": " . $prow['phone_home'];
+    }
 
-        if ($prow['phone_biz']) {
-            $patienttitle['phone_biz'] = xl("Work Phone").": " . $prow['phone_biz'];
-        }
+    if ($prow['phone_biz']) {
+        $patienttitle['phone_biz'] = xl("Work Phone").": " . $prow['phone_biz'];
     }
+}
```

Comment:
Your feature actually illustrates a potential minor bug.
At Administration->Globals->Features->Default Visit Category you can choose the default category. Note this is used in the new encounter GUI, when selecting this; however, it is not used here, which it likely should be.
For your purposes, would you be ok with a practice default (or even a user default, since easy to make the globals user specific)? Or do you need patient specific defaults?

Reply:
See above discussion in altering patient_data table.  We really need per patient default or some other way to categorize patients that gives a default appointment type.